core/templates/domain/topic/NewlyCreatedStoryObjectFactorySpec.ts

```diff
@@ -40,12 +40,21 @@ describe('Newly Created Story Object Factory', () => {
     expect(story.isValid()).toBe(false);
 
     story.title = 'storyName1';
-    expect(story.isValid()).toBe(true);
+    expect(story.isValid()).toBe(false);
 
     story.description = '';
     expect(story.isValid()).toBe(false);
 
     story.description = 'story description';
+    expect(story.isValid()).toBe(false);
+
+    story.urlFragment = '';
+    expect(story.isValid()).toBe(false);
+
+    story.urlFragment = 'story Fragment';
+    expect(story.isValid()).toBe(false);
+
+    story.urlFragment = 'story-fragment';
```

Comment:
In its entirety, this test only validates that the validation is correct for the url fragment. Maybe update it to be more thorough and ensure that it checks for the other fields being valid, too.